### PR TITLE
TESTS: Fix Race Condition in Temp Path Creation

### LINKED ISSUE #33344: [CI] LuceneChangesSnapshotTests#testUpdateAndReadChangesConcurrently fails
`LuceneChangesSnapshotTests#testUpdateAndReadChangesConcurrently` fails occasionally on master and 6x. I reproduce locally with the following seed (the failure doesn't occur on each run though but I was able to reproduce after a few ones):
```
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+release-tests/1005/console

./gradlew :server:test \
  -Dtests.seed=95ED5AB5FEF63FF6 \
  -Dtests.class=org.elasticsearch.index.engine.LuceneChangesSnapshotTests \
  -Dtests.method="testUpdateAndReadChangesConcurrently" \
  -Dtests.security.manager=true \
  -Dbuild.snapshot=false \
  -Dtests.jvm.argline="-Dbuild.snapshot=false" \
  -Dtests.locale=bg \
  -Dtests.timezone=Pacific/Enderbury \
  -Dcompiler.java=10 \
  -Druntime.java=8
```

### PR DESCRIPTION
**Pull request by original-brownbear**
* Calling `createTempDir` concurrently here in
the `Follower`s causes collisions at times
which lead to `createEngine` throwing because
of unexpected files in the newly created temp
dir
   * Fixed by creating all temp dirs in the main test thread
* closes #33344

### CODE REVIEW

**Review by jimczi — APPROVED** [review]
LGTM

File: server/src/test/java/org/elasticsearch/index/engine/LuceneChangesSnapshotTests.java [hunk 1]
```diff
@@ -261,7 +264,7 @@ void pullOperations(Engine follower) throws IOException {
         @Override
         public void run() {
             try (Store store = createStore();
-                 InternalEngine follower = createEngine(store, createTempDir())) {
+                 InternalEngine follower = createEngine(store, tempDir)) {
```
Comment (dnhatn) [inline server/src/test/java/org/elasticsearch/index/engine/LuceneChangesSnapshotTests.java]: nit: we should rename `tempDir` to `translogPath`

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks for fixing this @original-brownbear.

**Comment by original-brownbear** [comment]
@jimczi @dnhatn thanks you two! Will merge  once green :)

**Pull request merged**